### PR TITLE
Reusable-normal-button-component-#23

### PR DESCRIPTION
**Pull request by philipNgungi037**
# Description

A reusable normal button component that can be reused all-over the in App.
Fixes https://github.com/italanta/elewa-website/issues/23

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Screenshot (optional)
button

## How Has This Been Tested?
No tests have been run, however; I am able to see my button as per design descriptions

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/elewa-website/src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-<!-- Render the elewa website header component before the router outlet and at the top of the main app component so that the header can be displayed at the top of every page / route of the main application -->
-<elewa-website-header></elewa-website-header>
 
-<router-outlet></router-outlet>
+<elewa-website-root></elewa-website-root>
+
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/app/app.component.html:4]: ?? The root component is only called in the index.html file

File: apps/elewa-website/src/app/app.component.ts [hunk 2]
```diff
@@ -7,4 +7,12 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'elewa-website';
+
+  /*  Button function; when button is clicked, an empty/void event is sent  */
+  
+  Click(): void {
+    /* an empty output. */
+
```
Comment (LemmyMwaura) [inline apps/elewa-website/src/app/app.component.ts:15]: This shouldn't be here

File: libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { ButtonData } from '@elewa-website/models/schema/ui/buttons';
+
+@Component({
+  selector: 'elewa-website-elewa-normal-button',
+  templateUrl: './elewa-normal-button.component.html',
+  styleUrls: ['./elewa-normal-button.component.scss'],
+})
+export class ElewaNormalButtonComponent {
+
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts:10]: JsDoc comments are added above the properties/methods.

File: libs/models/schema/ui/buttons/src/lib/button.interface.ts [hunk 4]
```diff
@@ -1,9 +1,11 @@
 export interface ButtonData {
-  text: string;
+    text: string
+  
+    bgColor?: string; /* Background colour */
+    color?: string; /* text color */
+    
+    hoverBgColor?: string;
+    hoverColor?: string;
```
Comment (LemmyMwaura) [inline libs/models/schema/ui/buttons/src/lib/button.interface.ts:8]: Why this change.

File: tsconfig.base.json [hunk 5]
```diff
@@ -15,17 +15,20 @@
     "skipDefaultLibCheck": true,
     "baseUrl": ".",
     "paths": {
-      "@elewa-website/elements/layout/buttons": [
-        "libs/elements/layout/buttons/src/index.ts"
-      ],
```
Comment (LemmyMwaura) [inline tsconfig.base.json:20]: This removes the aliased path to the buttons module from the codebase

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: apps/elewa-website/src/app/app.component.ts (hunk 2 above)
Comment (philipNgungi037) [reply apps/elewa-website/src/app/app.component.ts:15]: Am suppose to create new lib pages/home and add the click event there, or where am i going wrong?
@LemmyMwaura

**Comment by LemmyMwaura** [comment]
Closing as this was resolved
